### PR TITLE
fix: morphmath.angle_3points could not cope with collinear vectors

### PR DESCRIPTION
**Pull request by juanchopanza**
Add new tests for pi radians and zero angles and change the
implementation of angle calculation to a more robust one
using atan2 instead of acos..

### CODE REVIEW

File: neurom/analysis/morphmath.py [hunk 1]
```diff
@@ -83,12 +83,11 @@ def angle_3points(p0, p1, p2):
...
     Returns:
         Angle in radians between (p1-p0) and (p2-p0).
-        nan if p0==p1 or p0==p2.
+        0.0 if p0==p1 or p0==p2.
     '''
     vec1 = vector(p1, p0)
     vec2 = vector(p2, p0)
-    return math.acos(np.dot(vec1, vec2) /
-                     (np.linalg.norm(vec1) * np.linalg.norm(vec2)))
+    return math.atan2(np.linalg.norm(np.cross(vec1, vec2)), np.dot(vec1, vec2))
```
Comment (mgeplf) [inline neurom/analysis/morphmath.py]: huh,  math.atan2(0., 0.) == 0.0
weird

File: neurom/analysis/tests/test_morphmath.py [hunk 2]
```diff
@@ -74,21 +74,82 @@ def test_point_dist():
...
-def test_angle_3points():
+def test_angle_3points_half_pi():
+    orig = (0.0, 0.0, 0.0)
     vec1 = (1.0, 0.0, 0.0)
-    vec2 = (0.0, 1.0, 0.0)
-    orig = (0.0,0.0,0.0)
-    angle=angle_3points(orig, vec1, vec2)
-    nt.ok_(angle==pi/2.0)
+    vec2 = (0.0, 2.0, 0.0)
+    nt.ok_(angle_3points(orig, vec1, vec2)==pi/2.0)
```
Comment (mgeplf) [inline neurom/analysis/tests/test_morphmath.py]: use nt.eq_ so it will give you the numbers if/when it fails?

File: neurom/analysis/tests/test_morphmath.py [hunk 3]
```diff
@@ -74,21 +74,82 @@ def test_point_dist():
...
+
+    vec1 = (0.0, 4.0, 0.0)
+    nt.ok_(angle_3points(orig, vec1, vec2)==pi/2.0)
+
+
+def test_angle_3points_quarter_pi():
+    orig = (0.0, 0.0, 0.0)
+    vec1 = (1.0, 0.0, 0.0)
+    vec2 = (2.0, 2.0, 0.0)
+    nt.assert_equal(angle_3points(orig, vec1, vec2), pi / 4.0)
```
Comment (mgeplf) [inline neurom/analysis/tests/test_morphmath.py]: I also prefer eq_ to assert_equal, as that actually comes from unittest via a weird set of imports, fyi

**Comment by mgeplf** [comment]
+2 from me

File: neurom/analysis/tests/test_morphmath.py (hunk 2 above)
Comment (juanchopanza) [inline neurom/analysis/tests/test_morphmath.py]: Will do. I've been using `nt.assert_equal` for that.

File: neurom/analysis/tests/test_morphmath.py (hunk 3 above)
Comment (juanchopanza) [inline neurom/analysis/tests/test_morphmath.py]: Any benefits in doing that?
